common/resource/resourceImpl.go

```diff
@@ -563,28 +577,71 @@ func (h *Impl) GetDispatcher() *yarpc.Dispatcher {
 	return h.dispatcher
 }
 
-// GetIsolationGroupState returns the isolationGroupState
+// GetIsolationGroupState returns the isolationGroupState or nil
 func (h *Impl) GetIsolationGroupState() isolationgroup.State {
 	return h.isolationGroups
 }
 
-// GetPartitioner returns the partitioner
+// GetPartitioner returns the partitioner or nil
 func (h *Impl) GetPartitioner() partition.Partitioner {
 	return h.partitioner
 }
 
+// GetIsolationGroupStore returns the isolation group configuration store or nil
+func (h *Impl) GetIsolationGroupStore() configstore.Client {
+	return h.isolationGroupConfigStore
+}
+
+// due to the config store being only available for some
+// persistence layers, *both* the configStoreClient and IsolationGroupState
+// will be optionally available
+func createConfigStoreOrDefault(
+	params *Params,
+	dc *dynamicconfig.Collection,
+) configstore.Client {
+
+	if params.IsolationGroupStore != nil {
+		return params.IsolationGroupStore
+	}
+	cscConfig := &csc.ClientConfig{
+		PollInterval:        dc.GetDurationProperty(dynamicconfig.IsolationGroupStateRefreshInterval)(),
+		UpdateRetryAttempts: dc.GetIntProperty(dynamicconfig.IsolationGroupStateUpdateRetryAttempts)(),
+		FetchTimeout:        dc.GetDurationProperty(dynamicconfig.IsolationGroupStateFetchTimeout)(),
+		UpdateTimeout:       dc.GetDurationProperty(dynamicconfig.IsolationGroupStateUpdateTimeout)(),
+	}
+	cfgStoreClient, err := configstore.NewConfigStoreClient(cscConfig, &params.PersistenceConfig, params.Logger, persistence.GlobalIsolationGroupConfig)
+	if err != nil {
+		// not possible to create the client under some persistence configurations, so this is expected
+		return nil
+	}
+	return cfgStoreClient
+}
+
 // Use the provided IsolationGroupStateHandler or the default one
-func ensureIsolationGroupStateHandlerOrDefault(
+// due to the config store being only available for some
+// persistence layers, *both* the configStoreClient and IsolationGroupState
+// will be optionally available
+func createIsolationGroupStateHandlerOrDefault(
```

Comment:
changing the name to `create...` to indicate that it's not guaranteed to not be nil. `ensure` has, in my experience mostly been used to do the opposite - to guarantee non-nil-ness